src/contract.rs

```diff
@@ -227,12 +237,27 @@ pub fn authenticated_queries<S: Storage, A: Api, Q: Querier>(
     })?)
 }
 
-/// This function just returns a constant 1:1 rate to uscrt, since that's the purpose of this
-/// contract.
-fn query_exchange_rate() -> QueryResult {
+fn query_exchange_rate<S: ReadonlyStorage>(storage: &S) -> QueryResult {
+    let config = ReadonlyConfig::from_storage(storage);
+    let constants = config.constants()?;
+
+    if constants.deposit_is_enabled || constants.redeem_is_enabled {
+        let rate: Uint128;
+        let denom: String;
+        // if token has more decimals than SCRT, you get magnitudes of SCRT per token
+        if constants.decimals >= 6 {
+            rate = Uint128(10u128.checked_pow((constants.decimals - 6).into()).unwrap());
+            denom = "SCRT".to_string();
+        // if token has less decimals, you get magnitudes token for SCRT
+        } else {
+            rate = Uint128(10u128.checked_pow((6 - constants.decimals).into()).unwrap());
+            denom = constants.symbol;
+        }
```

Comment:
[Arithmetic overflow checks are globally enabled in this repo](https://github.com/enigmampc/snip20-reference-impl/blob/a91bfc3c38b13ed30aec07256049719812d3bea9/Cargo.toml#L27) so you can avoid the explicit "checked unwrap" (and some additional aesthetic tweaks):
```rust
rate = Uint128(10_u128.pow(constants.decimals as u128 - 6));
```